client.go

```diff
@@ -445,6 +450,9 @@ func baseRetryPolicy(resp *http.Response, err error) (bool, error) {
 			}
 
 			// Don't retry if the error was due to TLS cert verification failure.
+			if notTrustedErrorRe.MatchString(v.Error()) {
```

Comment:
Did you observe some specific condition where the existing `UnknownAuthorityError` check did not catch this? I'd prefer if we kept unrelated fixes like this in separate PRs.

Reply:
Yep, this test was failing. https://github.com/hashicorp/go-retryablehttp/blob/master/client_test.go#L577 (specifically with "expected 1 attempt, got 5")

I should probably also update the test a bit more carefully though, since it's probably no longer an x509 error

Reply:
Never mind, it still is. The full error (in string format) is
`2022/04/13 13:31:14 [ERR] GET https://127.0.0.1:58511 request failed: Get "https://127.0.0.1:58511": x509: “Acme Co” certificate is not trusted`

Reply:
Strange! In that case, this is totally fine. I wonder if it would be more comprehensive to just replace our origina UnknownAuthority check with an `/x509:/` regex or something like that. There are [a lot of permanent errors](https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/crypto/x509/verify.go) that appear to fall into that category. Either way - if this unblocks you, I'm fine with adding just the one extra check for now, and revisiting this later if needed.

Reply:
I'll just leave it, and we can revisit later if we ever need to.